src/lib/comms/sol-network-impl-linux.c

```diff
@@ -574,19 +577,19 @@ sol_network_link_set_status(uint16_t link_index, unsigned int changes, unsigned
 
     if (sendmsg(network->nl_socket, (struct msghdr *)&msg, 0) <= 0) {
         SOL_WRN("Failed on send message to set link up");
-        return false;
+        return -errno;
```

Comment:
Careful with those, even the if the function uses errno to set the error, SOL_WRN (and related) may re-set errno to some other value. In these cases you need to store errno to a temporary variable before you call anything else, and then return (or set back) that value.